components/scream/src/dynamics/homme/physics_dynamics_remapper.hpp

```diff
@@ -166,13 +163,71 @@ class PhysicsDynamicsRemapper : public AbstractRemapper<RealType>
 
   template <typename ScalarT, typename MT, typename VT1, typename VT2>
   KOKKOS_FUNCTION
-  void local_remap_bwd_2d (const MT& team, const int num_cols, const VT1 lid2elgp, const VT2 p2d) const;
+  void local_remap_bwd_2d (const MT& team, const int num_fields, const VT1 lid2elgp, const VT2 p2d) const;
 
   template <typename ScalarT, typename MT, typename VT1, typename VT2>
   KOKKOS_FUNCTION
-  void local_remap_bwd_3d (const MT& team, const int num_cols, const VT1 lid2elgp, const VT2 p2d) const;
+  void local_remap_bwd_3d (const MT& team, const int num_fields, const VT1 lid2elgp, const VT2 p2d) const;
 
   void create_p2d_map ();
+
+  template<typename ScalarT, int N>
+  KOKKOS_FUNCTION
+  Unmanaged<typename KokkosTypes<device_type>::template view_ND<ScalarT,N>>
+  reshape (Pointer& p, const Dims& dims) const {
+    using uview_nd = Unmanaged<typename KokkosTypes<device_type>::template view_ND<ScalarT,N>>;
+
+    switch (dims.size) {
+      case 1:
+      {
```

Comment:
MInor: you only need braces inside a `case` if you declare local variables, otherwise you don't. No braces would get rid of a few lines (and help fit more code in one screen).